### ISSUE TITLE
E231 (probably others): python3.12+ f-string false positives

### ISSUE THREAD

**Issue by asottile** [PyCQA/pycodestyle#1145]
now that f-strings are tokenized there's quite a few false positives.  I haven't done too much digging yet but this seems to be easiest one to reproduce:

```python
f'hello:{hi}'
```

```
t.py:1:8: E231 missing whitespace after ':'
```

**Comment by asottile** [comment]
via #1148
